languages/python/django-oso/django_oso/partial.py

```diff
@@ -230,11 +232,16 @@ def not_expr(self, expr: Expression):
         self.filter &= ~fb.finish()
 
     def finish(self):
-        # For every subquery, finish off by checking these are non-empty
+        """For every subquery, construct a filter to make sure the result set is non-empty"""
+        if len(self.variables) == 0:
+            return self.filter
+        objects = self.model.objects.all()
         for subq in self.variables.values():
             filtered = subq.model.objects.filter(subq.finish()).values("pk")
             exists = Exists(filtered)
-            self.filter = exists & self.filter  # This _has_ to be this way around
+            name = f"{self.name}__exists"
+            objects = objects.annotate(**{name: exists}).filter(**{name: True})
```

Comment:
This is cute. For anyone happening upon this line in the future: https://docs.djangoproject.com/en/2.2/ref/models/expressions/#filtering-on-a-subquery-expression

Reply:
Will add as a comment

Reply:
@samscott89 ping

Reply:
done